verilated.mk: Apple clang does not support -o flag in conjunction with PCH

Using latest git verilator, on MacOS 26.3.1. When compiling some verilated sources I get:

```
/usr/bin/c++ -Os  -I.  -MMD -I/usr/local/include -I/usr/local/include/vltstd -DVERILATOR=1 -DVM_COVERAGE=0 -DVM_SC=0 -DVM_TIMING=0 -DVM_TRACE=0 -DVM_TRACE_FST=0 -DVM_TRACE_VCD=0 -DVM_TRACE_SAIF=0        Vtb__pch.h.fast.gch -c -o Vtb.o Vtb.cpp
clang++: error: cannot specify -o when generating multiple output files
make[1]: *** [/usr/local/include/verilated.mk:283: Vtb.o] Error 1
```

Problematic lines are here in `verilated.mk`:

https://github.com/verilator/verilator/blob/4b53f5f978f9a48d12c597d8cd66fec9606d8c9b/include/verilated.mk.in#L282-L286

Apple clang rejects the `-o` flag when PCH is used. Note that native GCC isn't available on MacOS and `gcc` is just symlinked to `clang`.

Since this is just a `%.o: %.cpp` pattern the `-o` flag is unnecessary. As a workaround I've removed `-o -@` from lines 283 and 286 in that Makefile; if I do this then everything builds and simulates fine for me. However I'm not sure if this is an appropriate fix for all platforms.

--- a/docs/CONTRIBUTORS
+++ b/docs/CONTRIBUTORS
@@ -60,6 +60,7 @@ Driss Hafdi
 Edgar E. Iglesias
 Eric Müller
 Eric Rippey
+Eunseo Song
 Ethan Sifferman
 Eyck Jentzsch
 Fabian Keßler-Schulz

--- a/include/verilated.mk.in
+++ b/include/verilated.mk.in
@@ -280,10 +280,10 @@ ifneq ($(VM_DEFAULT_RULES),0)
 	$(OBJCACHE) $(CXX) $(OPT_FAST) $(CXXFLAGS) $(CPPFLAGS) -c -o $@ $<
 
   $(VK_OBJS_FAST): %.o: %.cpp $(VK_PCH_H).fast.gch
-	$(OBJCACHE) $(CXX) $(OPT_FAST) $(CXXFLAGS) $(CPPFLAGS) $(VK_PCH_I_FAST) -c -o $@ $<
+	$(OBJCACHE) $(CXX) $(OPT_FAST) $(CXXFLAGS) $(CPPFLAGS) $(VK_PCH_I_FAST) -c $<
 
   $(VK_OBJS_SLOW): %.o: %.cpp $(VK_PCH_H).slow.gch
-	$(OBJCACHE) $(CXX) $(OPT_SLOW) $(CXXFLAGS) $(CPPFLAGS) $(VK_PCH_I_SLOW) -c -o $@ $<
+	$(OBJCACHE) $(CXX) $(OPT_SLOW) $(CXXFLAGS) $(CPPFLAGS) $(VK_PCH_I_SLOW) -c $<
 
   $(VK_GLOBAL_OBJS): %.o: %.cpp
 	$(OBJCACHE) $(CXX) $(OPT_GLOBAL) $(CXXFLAGS) $(CPPFLAGS) -c -o $@ $<

